modules/videoio/src/cap_mjpeg_encoder.cpp

```diff
@@ -331,9 +331,14 @@ class mjpeg_buffer_keeper
         }
 
         //bits == 0 means that last element shouldn't be used.
-        m_output_buffer[m_data_len++] = currval;
-
-        m_last_bit_len = -bits;
+        if (bits != 0) {
```

Comment:
If bits != 0 than currval contains some useful bits. We need to increase m_data_len and add the bits to the last item of m_output_buffer.
If bits == 0 than currval doesn't contain useful bits